ci-operator/step-registry/sandboxed-containers-operator/create-prowjob/sandboxed-containers-operator-create-prowjob-commands.sh

```diff
@@ -0,0 +1,418 @@
+if [[ "${INSTALL_KATA_RPM}" != "true" && "${INSTALL_KATA_RPM}" != "false" ]]; then
+    echo "ERROR: INSTALL_KATA_RPM should be 'true' or 'false', got: ${INSTALL_KATA_RPM}"
+    exit 1
+fi
+
+# Kata RPM version (includes OCP version)
+if [[ "${INSTALL_KATA_RPM}" == "true" ]]; then
+    KATA_RPM_VERSION="${KATA_RPM_VERSION:-3.17.0-3.rhaos4.19.el9}"
+else
+    KATA_RPM_VERSION="${KATA_RPM_VERSION:-}"
+fi
+
+# Prow Configuration
+PROW_RUN_TYPE="${PROW_RUN_TYPE:-candidate}"
+# Validate PROW_RUN_TYPE
+if [[ "${PROW_RUN_TYPE}" != "candidate" && "${PROW_RUN_TYPE}" != "release" ]]; then
+    echo "ERROR: PROW_RUN_TYPE should be 'candidate' or 'release', got: ${PROW_RUN_TYPE}"
+    exit 1
+fi
+
+# After the tests finish, wait before killing the cluster
+SLEEP_DURATION="${SLEEP_DURATION:-0h}"
+# Validate SLEEP_DURATION format (0-12 followed by 'h')
+if ! [[ "${SLEEP_DURATION}" =~ ^(1[0-2]|[0-9])h$ ]]; then
+    echo "ERROR: SLEEP_DURATION must be a number between 0-12 followed by 'h' (e.g., 2h, 8h), got: ${SLEEP_DURATION}"
+    exit 1
+fi
+
+# test is Pre-GA for brew builds or GA for operators/rpms already on OCP
+# this triggers the mirror redirect install, creating brew & trustee catsrc,
+TEST_RELEASE_TYPE="${TEST_RELEASE_TYPE:-Pre-GA}"
+# Validate TEST_RELEASE_TYPE
+if [[ "${TEST_RELEASE_TYPE}" != "Pre-GA" && "${TEST_RELEASE_TYPE}" != "GA" ]]; then
+    echo "ERROR: TEST_RELEASE_TYPE should be 'Pre-GA' or 'GA', got: ${TEST_RELEASE_TYPE}"
+    exit 1
+fi
+
+# Let the tests run for this many minutes before killing the cluster and interupting the test
+TEST_TIMEOUT="${TEST_TIMEOUT:-90}"
+# Validate TEST_TIMEOUT is numeric
+if ! [[ "${TEST_TIMEOUT}" =~ ^[0-9]+$ ]]; then
+    echo "ERROR: TEST_TIMEOUT should be numeric, got: ${TEST_TIMEOUT}"
+    exit 1
+fi
+
+
+
+# Catalog Source Configuration
+echo "Configuring catalog sources..."
+
+# Set catalog source variables based on TEST_RELEASE_TYPE
+if [[ "${TEST_RELEASE_TYPE}" == "Pre-GA" ]]; then
+    # OSC Catalog Configuration - get latest or use provided
+    if [[ -z "${OSC_CATALOG_TAG:-}" ]]; then
+        OSC_CATALOG_TAG=$(get_latest_osc_catalog_tag)
+    else
+        echo "Using provided OSC_CATALOG_TAG: ${OSC_CATALOG_TAG}"
+    fi
+
+    CATALOG_SOURCE_IMAGE="${CATALOG_SOURCE_IMAGE:-quay.io/redhat-user-workloads/ose-osc-tenant/osc-test-fbc:${OSC_CATALOG_TAG}}"
+    CATALOG_SOURCE_NAME="${CATALOG_SOURCE_NAME:-brew-catalog}"
+
+    # Trustee Catalog Configuration
+    # Convert OCP version for Trustee catalog naming
+    OCP_VER=$(echo "${OCP_VERSION}" | tr '.' '-')
+    subfolder=""
+    if [[ "${OCP_VER}" == "4-16" ]]; then
+        subfolder="trustee-fbc/"
+    fi
+    # Get latest Trustee catalog tag with page limit safety
+    TRUSTEE_REPO_NAME="${subfolder}trustee-fbc-${OCP_VER}"
+    TRUSTEE_CATALOG_REPO="quay.io/redhat-user-workloads/ose-osc-tenant/${TRUSTEE_REPO_NAME}"
+
+    APIURL="https://quay.io/api/v1/repository/redhat-user-workloads/ose-osc-tenant/${TRUSTEE_REPO_NAME}"
```

Comment:
How about using main "APIURL" and then only per-project suffix embedded in functions? Or at least unifying the behaviour of `get_latest_osc_catalog_tag` vs. `get_latest_trustee_catalog_tag` (one uses local variable, other requires one to configure env variable).